cs/src/core/Index/FasterLog/ILogCommitManager.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.IO;
+
+namespace FASTER.core
+{
+    /// <summary>
+    /// Log commit manager
+    /// </summary>
+    public interface ILogCommitManager
+    {
+        /// <summary>
+        /// Perform (synchronous) commit with specified metadata
+        /// </summary>
+        /// <param name="address">Address committed until (for information only, not necessary to persist)</param>
+        /// <param name="commitMetadata">Commit metadata</param>
+        void Commit(long address, byte[] commitMetadata);
```

Comment:
I am wondering if `byte[]` is too low level of an abstraction. Perhaps an interface with a custom serialization/deserialization method is a better choice?

Reply:
I thought of that and even tried to prototype it but it was not as convenient in the end because the user had to implement too many callbacks. This can be revisited though.